File: Stage18/Python/Problem2108.py
import sys
input = sys.stdin.readline

def main():
    N = int(input())
    nums = []
    freq = [0] * 8001

    _sum = 0
    for _ in range(N):
        x = int(input())
        nums.append(x)
        _sum += x
        freq[x + 4000] += 1

    avg = round(_sum / N)

    nums.sort()
    mid = nums[N // 2]

    max_freq = max(freq)
    count = 0
    mode = 0
    for i in range(8001):
        if freq[i] == max_freq:
            mode = i - 4000
            count += 1
            if count == 2:
                break

    range_val = nums[-1] - nums[0]

    print(avg)
    print(mid)
    print(mode)
    print(range_val)

if __name__ == "__main__":
    main()

File: Stage18/Java/Problem2108.java
package Stage18.Java;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;

public class Problem2108 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        int N = Integer.parseInt(br.readLine());

        int[] arr = new int[N];
        int[] freq = new int[8001];
        int sum = 0;

        for (int i = 0; i < N; i++) {
            int x = Integer.parseInt(br.readLine());
            arr[i] = x;
            sum += x;
            freq[x + 4000]++;
        }

        int avg = (int) Math.round((double) sum / N);

        Arrays.sort(arr);
        int mid = arr[N / 2];

        int maxFreq = 0;
        for (int f : freq) maxFreq = Math.max(maxFreq, f);

        int mode = 0;
        int countFreq = 0;
        for (int i = 0; i < 8001; i++) {
            if (freq[i] == maxFreq) {
                mode = i - 4000;
                countFreq++;
                if (countFreq == 2) break;
            }
        }

        int range = arr[N - 1] - arr[0];

        sb.append(avg).append("\n");
        sb.append(mid).append("\n");
        sb.append(mode).append("\n");
        sb.append(range);

        System.out.println(sb);
        br.close();
    }
}
